test/jdk/java/net/httpclient/PlainConnectionLockTest.java

```diff
@@ -0,0 +1,382 @@
+            http1Server.stop();
+        }
+        if (https1Server != null) {
+            System.out.println("Stopping HTTPS server " + https1Server.getAddress());
+            https1Server.stop();
+        }
+        if (serverExecutor != null) {
+            System.out.println("Closing server executor");
+            if (successfulCompletion) {
+                serverExecutor.close();
+            } else {
+                // server handlers may be wedged.
+                serverExecutor.shutdownNow();
+            }
+        }
+        requestSemaphore = null;
+        responseSemaphore = null;
+        serverExecutor = null;
+        http1Server = null;
+        https1Server = null;
+        http1URI = null;
+        https1URI = null;
+        System.out.println("done\n");
+    }
+
+    @Test
+    void sendManyHttpRequestsNoShutdown() throws Exception {
+        try {
+            sendManyRequests(http1URI, MANY, false);
+        } catch (Throwable t) {
+            t.printStackTrace(System.out);
+            throw t;
+        }
+    }
+
+    @Test
+    void sendManyHttpRequestsShutdownNow() throws Exception {
+        try {
+            sendManyRequests(http1URI, MANY, true);
+        } catch (Throwable t) {
+            t.printStackTrace(System.out);
+            throw t;
+        }
+    }
+
+    @Test
+    void sendManyHttpsRequestsNoShutdown() throws Exception {
+        try {
+            sendManyRequests(https1URI, MANY, false);
+        } catch (Throwable t) {
+            t.printStackTrace(System.out);
+            throw t;
+        }
+    }
+
+    @Test
+    void sendManyHttpsRequestsShutdownNow() throws Exception {
+        try {
+            sendManyRequests(https1URI, MANY, true);
+        } catch (Throwable t) {
+            t.printStackTrace(System.out);
+            throw t;
+        }
+    }
+
+    private static void throwCause(CompletionException x) throws Exception {
+        var cause = x.getCause();
+        if (cause instanceof Exception ex) throw ex;
+        if (cause instanceof Error err) throw err;
+        throw x;
+    }
+
+    static final long start = System.nanoTime();
+    public static String now() {
+        long now = System.nanoTime() - start;
+        long secs = now / 1000_000_000;
+        long mill = (now % 1000_000_000) / 1000_000;
+        long nan = now % 1000_000;
+        return String.format("[%d s, %d ms, %d ns] ", secs, mill, nan);
+    }
+
+    static Throwable getCause(Throwable exception) {
+        if (exception instanceof IOException) return exception;
+        if (exception instanceof CancellationException) return exception;
+        if (exception instanceof CompletionException) return getCause(exception.getCause());
+        if (exception instanceof ExecutionException) return getCause(exception.getCause());
+        return exception;
+    }
+
+    private synchronized void sendManyRequests(final String requestURI, final int many, boolean shutdown) throws Exception {
```

Comment:
When `shutdown = false`, we expect all request-response exchange to succeed with 200. In the context of the tested behaviour, which case does `shutdown = false` stress?